hivemind/moe/server/runtime.py

```diff
@@ -108,6 +99,17 @@ def run(self):
                 if not self.shutdown_trigger.is_set():
                     self.shutdown()
 
+    def process_batch(self, pool: TaskPoolBase, batch_index: int, *batch: torch.Tensor):
+        """process one batch of tasks from a given pool, return a batch of results"""
+        start = time()
+        outputs = pool.process_func(*batch)
+        batch_processing_time = time() - start
+        batch_size = outputs[0].size(0)
+        logger.debug(f"Pool {pool.name}: batch {batch_index} processed, size {batch_size}")
+        if self.stats_report_interval is not None:
+            self.stats_reporter.report_stats(pool.name, batch_size, batch_processing_time)
```

Comment:
I think it might lead to unintended "0 batches processed" log entries if the user overrides this without carefully considering the original function. Best to leave only the batch size computation inside the function and keep all the logging/time measuring logic outside the function

Reply:
good catch, fixed it now, please take another look